### PR TITLE
UIP-52 fix integration tests

### PR DESCRIPTION
**Pull request by briehl**
# STILL IN PROGRESS - DO NOT MERGE YET

# Description of PR purpose/changes

Fix up the broken integration tests. They've failed due to a combination of out-of-date dependencies, a bug or two, and inaccessible test narratives.

I've changed relevant narratives to be accessible (or at least viewable) by the `narrative_test` CI account. So they should all work ok with a `narrative_test` auth token, which should be a GH secret.

Here's the changes:
* updated to latest chromedriver and puppeteer (still the standard for installing headless chrome)
* fixed bug in public data view where the number of contigs and features for NCBI ref genomes wasn't being shown properly
* updated the narrative in the featureset viewer test to a new one owned by `narrative_test`
* made other narratives viewable by the `narrative_test` account (TODO: remake these under `narrative_test`)

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/UIP-52
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [n/a] (Python) I have run Ruff `format` and `check` on changed Python code manually or with a git precommit hook
- [x] Any dependent changes have been merged and published in downstream modules

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: kbase-extension/static/kbase/custom/custom.js [hunk 1]
```diff
@@ -145,19 +143,8 @@ define([
             options = defaultValue;
             defaultValue = undefined;
         }
-        let missing = false;
```
Comment (briehl) [inline kbase-extension/static/kbase/custom/custom.js:148]: This hunk of code here tests if the string passed in for processing is a number in various ways. We're using the [numeral](http://numeraljs.com/) package here, so just let it tell us. Also, the real bug was using `isNaN` on the `numeralValue` object, which was erroring. Not sure why this took so long to fail / be found.

File: scripts/run_tests.sh [hunk 2]
```diff
@@ -46,13 +46,13 @@ if [ "$run_docker" == 1 ]; then
         --template \
         "/kb/dev_container/narrative/src/config.json.templ:/kb/dev_container/narrative/kbase-extension/static/kbase/config/config.json" \
         kbase-narrative --no-browser --port=$JUPYTER_PORT 2>&1 | tee $OUTPUT_FILE &
-    bg_pid=$!
+    bg_pid=$(jobs -p)
```
Comment (briehl) [inline scripts/run_tests.sh:49]: Tests ongoing - I want to see how it behaves on GHA.
But when running on my local MacOS, `$bg_pid` had the pid of the `tee` process, not `kbase-narrative`. Running `jobs -p` fixes that.

File: scripts/run_tests.sh [hunk 3]
```diff
@@ -113,19 +113,22 @@ done
...
+#         done
+#     fi
+#     if [[ "$and_self" == true ]]; then
+#         echo "killing $pid"
+#         kill -9 "$pid"
+#     fi
+# }
+
+# echo "Killing process $bg_pid and children"
+pkill -P "$bg_pid"
```
Comment (briehl) [inline scripts/run_tests.sh:131]: We're already using `pgrep` in `kill_descendant_processes`, so `pkill` should be available and basically does the same thing. I think.

File: test/integration/specs/kbaseFeatureSet_data.json [hunk 4]
```diff
@@ -57,140 +57,42 @@
                             }
                         ]
                     },
-                    {
```
Comment (briehl) [inline test/integration/specs/kbaseFeatureSet_data.json:60]: These tests seemed to be effectively redundant with slightly different data. They may be relevant, and just generated by different apps, but as far as the viewer's concerned, they seem to be similar enough.

File: kbase-extension/static/kbase/custom/custom.js (hunk 1 above)
Comment (ialarmedalien) [reply kbase-extension/static/kbase/custom/custom.js:148]: I'm sure it's the only unfound bug in the codebase!

**Review by ialarmedalien — APPROVED** [review]

**Pull request merged**